packages/phenomic-plugin-renderer-react/src/render/server.js

```diff
@@ -0,0 +1,63 @@
+import path from "path"
+
+import React from "react"
+import ReactDOMServer from "react-dom/server"
+import { match, RouterContext } from "react-router"
+import { createRouteFromReactElement } from "react-router/lib/RouteUtils"
+import createURL from "phenomic-api-client/lib/url"
+
+import Provider from "../components/Provider"
+import createStore from "../shared/store"
+import performQuery from "../shared/performQuery"
+import QueryString from "../shared/QueryString"
+import renderHTML from "../server/renderHTML"
+
+const debug = require("debug")("phenomic:plugin:react")
+
+function getMatch({ routes, location }) {
+  return new Promise((resolve, reject) => {
+    match({ routes, location: `/${ location }` }, (error, redirectLocation, renderProps) => {
+      error
+      ? reject(error)
+      : resolve({ renderProps, redirectLocation })
+    })
+  })
+}
+
+function renderToString(store, { renderProps, redirectLocation }, renderHTML) {
+  const body = ReactDOMServer.renderToString(
+    <Provider fetch={ fetch } store={ store }>
```

Comment:
@bloodyowl can you explain me here how this fetch ("spec" fetch) can do his job here when Provider except another kind of fetch (the one that return .json()). I am probably too tired to understand this (I am finishing to add flow types everywhere in the codebase and this is the only issue left for me :) )